source/string_internal.hxx

```diff
@@ -91,13 +91,15 @@ enum EncodingTypeFlags : int {
 // A utility functor for appending encoded characters to a growing string
 
 template <typename C>
-class CharacterAdder : public std::unary_function<void, C>
+class CharacterAdder
 {
     static_assert(std::is_integral<C>::value, "CharacterAdder needs an integral type.");
 
 public:
-    CharacterAdder(std::basic_string<C>& result) : std::unary_function<void, C>(),
-                                                     m_destination(result) {}
+    using argument_type = void;
+    using result_type = C;
```

Comment:
These aren't used in the class. Do they satisfy some other contract that I am unaware of? I'm not seeing it inside my source either.

Reply:
This is exactly what inheriting from unary_function was doing, declaring those typedef. In the library itself they are not used. Even when the header is internal, I wanted to keep the existing behavior in the PR.

Reply:
fair enough. that's "that other contract". I'm not sure how widely the library is being used. I'd be willing to take them out and see what happens -- on the other hand, it'd kinder to leave them in and in the near future announce that in the somewhat less near future they'll go away. I was just following "best practices" à la Scott Meyers to bring std::unary_function in here at the time.

Reply:
Thank you for maintaining the library. Would you consider a 1.5.6 release for C++17 support? This would facilitate picking it up in our CI/CD servers. Thanks again.